### PR TITLE
Fix flux treatment of FFT renders to not double the Poisson noise

### PR DESCRIPTION
**Pull request by rmjarvis**
cf. Issue #419 

1. Switch to using the nominal flux for FFT renders, since we then add Poisson noise to the stamp image, which already does the work of Poisson-ifying the realized flux.
2. Switch the flux value used for all decisions (phot vs fft, faint vs not) to use the nominal flux, rather than the realized flux.
3. Changed the meaning of realized flux in the output file to be the actual number of photons that landed on the image, rather than the number of photons shot.  This way vignetting is consistently included in this definition regardless of whether the object used phot or fft.
4. Added phot_flux and fft_flux values in the output file.  Only one of these is non-zero, and it is the flux used for each rendering type.  So phot_flux is what used to be called realized_flux for photon shooting.  And fft_flux is what used to be called realized_flux for fft renders.  These each have slightly different meanings, but they are well-defined for each method.
5. Added tests of all these flux values in the output catalog.

### CODE REVIEW

File: imsim/stamp.py [hunk 1]
```diff
@@ -123,10 +123,20 @@ def setup(self, config, base, xsize, ysize, ignore, logger):
...
+        base['nominal_flux'] = self.nominal_flux
+        base['phot_flux'] = self.phot_flux
+        base['fft_flux'] = 0.  # For fft drawing, this will be updated in buildPSF.
+        base['realized_flux'] = 0.  # Will update this after drawImage using im.added_flux
+
+        # Check if the Poisson draw for the photon flux is 0.
+        if self.phot_flux == 0:
             # If so, we'll skip everything after this.
             # The mechanism within GalSim to do this is to raise a special SkipThisObject class.
             raise galsim.config.SkipThisObject('realized flux=0')
```
Comment (jchiang87) [inline imsim/stamp.py:142]: I guess this should be
```
raise galsim.config.SkipThisObject('phot_flux=0')
```

**Review by jchiang87 — APPROVED** [review]
lgtm.  One small comment about updating a log message.

**Comment by cwwalter** [comment]
Thanks Mike!

**Comment by jchiang87** [comment]
Should we go ahead and merge this?  Everything looks good to me.

**Comment by rmjarvis** [comment]
Sure.

**Pull request merged**